[Yeoh Han Yi] iP

yhanyi opened the pull request:
# DukePro

> "Your mind is for having ideas, not holding them." - David Allen ([source](link-to-source))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://github.com/yhanyi/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

yhanyi committed:
Level 0. Rename, Greet, Exit

yhanyi committed:
Level 1. Echo

yhanyi committed:
Level 2. Add, List

yhanyi committed:
A-Classes

yhanyi committed:
Level 3. Mark as Done

yhanyi committed:
A-Inheritance

yhanyi committed:
Level 4. ToDos, Events, Deadlines

yhanyi committed:
A-TextUiTesting

yhanyi committed:
A-Exceptions

yhanyi committed:
Level 5. Handle Errors

yhanyi committed:
A-Collections, Level 6. Delete

yhanyi committed:
Level 6. Delete

yhanyi committed:
Level 7. Save

yhanyi committed:
Merge branch 'branch-Level-7'

yhanyi committed:
Level 8. Dates and Times

yhanyi committed:
Level 8. Dates and Times

yhanyi committed:
Merge branch 'branch-Level-8'

yhanyi committed:
A-Packages

yhanyi committed:
Merge branch 'branch-A-Packages'

yhanyi committed:
A-MoreOOP

yhanyi committed:
Merge branch 'branch-A-MoreOOP'

yhanyi committed:
A-Enums

yhanyi committed:
Merge branch 'branch-A-Gradle'

yhanyi committed:
A-Gradle

yhanyi committed:
A-JUnit

yhanyi committed:
Merge branch 'branch-A-JUnit'

yhanyi committed:
A-JavaDoc

yhanyi committed:
A-CodingStandard

yhanyi committed:
Level 9. Find

yhanyi committed:
Merge branch 'branch-A-JavaDoc'

yhanyi committed:
Resolve merge conflict

yhanyi committed:
Resolve merge conflict

yhanyi committed:
A-CheckStyle

yhanyi committed:
Merge branch 'branch-A-CheckStyle'

yhanyi committed:
A-Varargs

yhanyi committed:
Merge branch 'branch-A-Varargs'

yhanyi committed:
Fixed Gradle issues

yhanyi committed:
Level-10

yhanyi committed:
Merge branch 'branch-Level-10'

yhanyi committed:
A-Assertions
Added assertions to ensure correctness of input preprocessing.
Changes were applied to Duke.java and Parser.java.

yhanyi committed:
A-CodeQuality
- Remove redundant variables specified in Duke.
- Refactor Task and subclasses to share DateTimeFormatters.

yhanyi committed:
A-Streams
Added usage of Java Streams into TaskList functionality.

yhanyi committed:
A-CI
Add Continuous Integration using GitHub Actions.

yhanyi committed:
Merge pull request #1 from yhanyi/branch-A-Assertions

A-Assertions

yhanyi committed:
Merge pull request #2 from yhanyi/branch-A-CodeQuality

A-CodeQuality

yhanyi committed:
Merge branch 'master' into branch-A-Streams

Resolved merge conflicts with prior pull requests.

yhanyi committed:
Merge pull request #3 from yhanyi/branch-A-Streams

A-Streams

yhanyi committed:
Updated yaml for GitHub Actions

yhanyi committed:
Merge pull request #4 from yhanyi/branch-A-CI

A-CI

yhanyi committed:
C-BetterSearch

yhanyi committed:
Merge branch 'branch-C-BetterSearch'

yhanyi committed:
Changes to the GUI

yhanyi committed:
Update ParserTest

yhanyi committed:
Update TaskListTest

yhanyi committed:
Merge branch 'branch-A-Increments'

yhanyi committed:
Add update functionality
Added update function allowing users to replace the task at a specified
index. Retains the mark status of the task that is replaced.

yhanyi committed:
Add tests for UpdateTask
Added additional test functions to test the update function.

yhanyi committed:
Add option to choose save file
Added function to allow users to specify a save file. This automatically
allows users to import/export saved files as well.

yhanyi committed:
Update file handling by Storage
Updated Storage class to handle files in normal and testing mode
similarly.

yhanyi committed:
Add tests for file switching function
Added additional tests to TaskListTest to test the file switching
function.

yhanyi committed:
Merge branch 'branch-C-Update'

yhanyi committed:
Merge branch 'branch-C-FlexibleDataSource'

Resolved merge conflict with branch-C-FlexibleDataSource.

jan-kai1 commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,358 @@
+                .map(line -> line.split("\\s*\\|\\s*"))
+                .forEach(parts -> {
+                    try {
+                        Task task = switch (parts[0]) {
```
I like the use of stream and splitting the string to parse out the type of command

jan-kai1 commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,358 @@
+    }
+
+    /**
+     * Filters and prints all tasks containing the input.
```
Could add @param to specify parameter

jan-kai1 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,150 @@
+    }
+
+    /**
+     * Sets up and enters testing environment.
```
Could specify what specific testing environment the method is used in

jan-kai1 commented on src/test/java/duke/parser/ParserTest.java:
```diff
@@ -0,0 +1,195 @@
+     * Tests the parser's response to the list command with tasks.
+     */
+    @Test
+    void testNonEmptyList() {
```
Good testing with many cases

jan-kai1 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,159 @@
+                case "delete":
+                    taskList.deleteTask(args);
+                    break;
+                case "todo":
```
cool use of not breaking to pass into a single method

jan-kai1 reviewed (commented):
LGTM, very minor details with test case description and documentation

accountexeregister commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,150 @@
+    private static final String DATA_DIR = "data/";
+    private static final String DEFAULT_FILE = "duke";
+    private static boolean isTestMode = false;
+    private static String currentFile = DEFAULT_FILE;
```
Great naming!

accountexeregister commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,150 @@
+            }
+            File file = new File(DATA_DIR + fileName);
+            if (!file.exists()) {
+                boolean success = file.createNewFile();
```
Should this be `isSuccessful` instead?

accountexeregister commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,150 @@
+    private static void deleteTestFile() {
+        File testFile = new File(DATA_DIR + getFullFileName());
+        if (testFile.exists()) {
+            boolean deleted = testFile.delete();
```
Should this be `isDeleted` instead?

accountexeregister commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,84 @@
+     *
+     * @return A string representation of the task
+     */
+    public String saveFormat() {
```
`saveFormat` might be interpreted as a void method. Should this be `getSaveFormat` instead since it returns a string?

accountexeregister reviewed (commented):
Overall, very well done. Code has good structure and and good naming. Test coverage is also quite comprehensive. I can clearly see a lot of effort being put into this.

yhanyi committed:
Add help command
Add "help" command for users to quickly access a helpsheet of available
commands within the chatbot.

yhanyi committed:
Add tests for help command
Added additional JUnit tests in both CLI and GUI for the help command.

yhanyi committed:
Merge branch 'branch-C-Help'

yhanyi committed:
Add archiving of tasks
Add archiving functionality where users can automatically archive their
current file to a new file postfixed with "archive".

yhanyi committed:
Added tests for archive function
Added additional JUnit tests for archive functionality.

yhanyi committed:
Merge branch 'branch-C-Archive'

yhanyi committed:
Add GUI title and Ui.png
Added the Duke title to the GUI window and screenshot of the UI to the
docs folder.

yhanyi committed:
Update README
Updated the README to document the features that Duke has to offer.

yhanyi committed:
Merge branch 'branch-A-UserGuide'
Merge the update user guide to master.

yhanyi committed:
Update gradle settings
Revert gradle settings to original.

yhanyi committed:
Rename all instances of Duke to SadCat
Update README docs and replace current Ui.png image.

yhanyi committed:
Rename all instances of Duke to SadCat
Renamed all instances of Duke to SadCat to finish renaming requirement
for Level-0.

yhanyi committed:
Update miscellaneous files
Update remaining miscellaneous files to reflect change in product name.

yhanyi committed:
Merge branch 'branch-A-Personality'
Merge branch for renaming product into master with no fast forwarding.

yhanyi committed:
Update gradle configs
Revert gradle configurations to default.

yhanyi committed:
Update JavaDocs comments
Updated JavaDocs comments to follow specified phrasing standards.

yhanyi committed:
Update TaskList switch block
Update TaskList to reduce the length of the switch block in creating a
task.

yhanyi committed:
Add Task factory methods
Added static factory methods to Task to create specific tasks.

yhanyi committed:
Include gradle configs for local testing
Included gradle settings and configurations used for local testing on
MacOS with ARM architecture.

yhanyi committed:
Update input handler in Parser
Abstracted the switch block of the input handlers in Parser into its own
method.

yhanyi committed:
Merge branch 'branch-MiscFixes'
Changes to Parser and TaskList and Tasks to reduce method lengths.

yhanyi committed:
Remove unnecessary uses of "this"
Removed unnecessary uses of "this" to follow coding conventions and
updated protected static final variable.

yhanyi committed:
Merge branch 'branch-MiscFixes'
Merge branch-MiscFixes with code cleanups.
